substrate/client/cli/src/commands/generate_node_key.rs

```diff
@@ -58,9 +74,33 @@ impl GenerateNodeKeyCmd {
 			array_bytes::bytes2hex("", secret).into_bytes()
 		};
 
-		match &self.file {
-			Some(file) => fs::write(file, file_data)?,
-			None => io::stdout().lock().write_all(&file_data)?,
+		match (&self.file, &self.base_path, self.default_base_path) {
+			(Some(file), None, false) => fs::write(file, file_data)?,
+			(None, Some(_), false) | (None, None, true) => {
+				let chain_spec = cli.load_spec(self.chain.as_deref().unwrap_or(""))?;
+
+				let network_path = build_network_key_dir_or_default(
+					self.base_path.clone().map(BasePath::new),
+					&chain_spec,
+					cli,
+				);
+
+				fs::create_dir_all(network_path.as_path())?;
+
+				let key_path = network_path.join(NODE_KEY_ED25519_FILE);
+				if key_path.exists() {
+					eprintln!("Skip generation, a key already exists in {:?}", key_path);
+					return Ok(());
```

Comment:
IMO should return an error to let the program return an error to make people "aware".

Reply:
Done.